src/Compilers/VisualBasic/Portable/Compilation/VisualBasicCompilation.vb

```diff
@@ -2231,7 +2231,8 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 SynthesizedMetadataCompiler.ProcessSynthesizedMembers(Me, moduleBeingBuilt, cancellationToken)
             Else
                 ' start generating PDB checksums if we need to emit PDBs
-                If emittingPdb AndAlso Not CreateDebugDocuments(moduleBeingBuilt.DebugDocumentsBuilder, moduleBeingBuilt.EmbeddedTexts, diagnostics) Then
+                If (emittingPdb OrElse moduleBeingBuilt.EmitOptions.EmitTestCoverageData) AndAlso
```

Comment:
When you say `/instrument:TestCoverage`, in addition to instrumenting each method, compiler also stores span information for all the source files that are part of the compilation in a resource in the instrumented binary. This span information includes a fingerprint (or hash) for each source file. This fingerprint was being being correctly computed and embedded in instrumented C# binaries but was missing in VB.

I believe the reason debug documents are required in this situation is because that is what we are using to compute the fingerprints  (i.e. instrumentation and pdbs share the same file fingerprints). As mentioned above, we were already doing this in C# and I simply replicated [this check](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Compilation/CSharpCompilation.cs,2322) in the C# code that was missing in VB to get this working. However, if you are asking whether creating debug documents when we are not emitting pdbs can have other side effects, then @tmat is probably the expert here...